### PR TITLE
feat: --reachable-vulns supports --all-projects flag in CLI

### PR DESCRIPTION
**Pull request by AlbertKogan**
#### What does this PR do?
Adding support of `--all-projects` flag altogether with `--reachable-vulns` for `snyk test` CLI command

#### How should this be manually tested?
Run in CLI `snyk test --all-projects --reachable-vulns`

### CODE REVIEW

File: src/lib/options-validator.ts [hunk 1]
```diff
@@ -6,12 +6,13 @@ import * as reachableVulns from './reachable-vulns';
 export async function validateOptions(
   options: (Options & TestOptions) | (Options & MonitorOptions),
   packageManager?: SupportedPackageManagers,
-) {
+): Promise<void> {
   if (options.reachableVulns) {
-    if (!packageManager) {
+    // Throwing error only in case when both packageManager and allProjects not defined
+    if (!packageManager && !options.allProjects) {
```
Comment (andremidea) [inline src/lib/options-validator.ts:12]: The only reason why it wouldn't work was this validation?

Comment (andremidea) [reply src/lib/options-validator.ts:12]: I asked that because I don't understand the implication of the `allProjects` flag

Comment (AlbertKogan) [reply src/lib/options-validator.ts:12]: @andremidea This check prevented workflow and passing `reachableVulns` option to `snyk-mvn-plugin` and after to `java-call-graph-builder` module where actual work done. This option will be ignoring in other plugins.

File: src/lib/reachable-vulns.ts [hunk 2]
```diff
@@ -29,10 +29,13 @@ export function serializeCallGraphWithMetrics(
 }
 
 export async function validatePayload(
-  packageManager: SupportedPackageManagers,
+  packageManager: SupportedPackageManagers | null,
   org: any,
 ): Promise<boolean> {
-  if (!REACHABLE_VULNS_SUPPORTED_PACKAGE_MANAGERS.includes(packageManager)) {
+  if (
+    packageManager &&
```
Comment (darmalovan) [inline src/lib/reachable-vulns.ts:36]: I'm afraid of `null` package manager that are coming from other places bypassing this logic.
Maybe worth also passing options.allProjects as a parameter to the function?

File: src/lib/options-validator.ts [hunk 3]
```diff
@@ -6,12 +6,13 @@ import * as reachableVulns from './reachable-vulns';
...
       throw new Error('Could not determine package manager');
     }
     const org = options.org || config.org;
-    await reachableVulns.validatePayload(packageManager, org);
+    await reachableVulns.validatePayload(packageManager ?? null, org);
```
Comment (darmalovan) [inline src/lib/options-validator.ts:16]: maybe the `null` part can be set as a default value, e.g.
`packageManager: SupportedPackageManagers | null = null`

File: src/lib/reachable-vulns.ts (hunk 2 above)
Comment (AlbertKogan) [reply src/lib/reachable-vulns.ts:36]: @darmalovan Tried to avoid changing of function's parameters. Can you elaborate a bit more about your concerns?

File: src/lib/options-validator.ts [hunk 4]
```diff
@@ -5,10 +5,11 @@ import * as reachableVulns from './reachable-vulns';
 
 export async function validateOptions(
   options: (Options & TestOptions) | (Options & MonitorOptions),
-  packageManager?: SupportedPackageManagers,
-) {
+  packageManager: SupportedPackageManagers | null = null,
```
Comment (dkontorovskyy) [inline src/lib/options-validator.ts:8]: `packageManager: SupportedPackageManagers | null = null` means packageManager can be null when before it was defaulting to `undefined`, cos of `?` why this change is crucial?

Comment (AlbertKogan) [reply src/lib/options-validator.ts:8]: @daniel The intend here is to prevent changing order of parameters for this function `await reachableVulns.validatePayload(packageManager, org);`. So if you want to keep this order, you need to provide value, because TS requires non optional parameters first

File: src/lib/reachable-vulns.ts (hunk 2 above)
Comment (darmalovan) [reply src/lib/reachable-vulns.ts:36]: My worry is that in the future a packageManager can be propagated here with a `null` value, without the allProjects parameter, and it will be considered valid.

Comment (AlbertKogan) [reply src/lib/reachable-vulns.ts:36]: Hmm, but why without `allProjects`? We not modifying `options` object, so this parameter should be there

Comment (AlbertKogan) [reply src/lib/reachable-vulns.ts:36]: Ah, you mean here in this function. Get it

Comment (AlbertKogan) [reply src/lib/reachable-vulns.ts:36]: But we have this check in `validateOptions` function
``` javascript
if (!packageManager && !options.allProjects) {
    throw new Error('Could not determine package manager');
}
```

Comment (darmalovan) [reply src/lib/reachable-vulns.ts:36]: `validatePayload` is a public function that can be used in the future from other callsites

Comment (AlbertKogan) [reply src/lib/reachable-vulns.ts:36]: Ok, so maybe it worth to pass `options` object?

Comment (darmalovan) [reply src/lib/reachable-vulns.ts:36]: Sounds good :)
